hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreCompactor.java

```diff
@@ -74,14 +74,15 @@
 
   int immutCellsNum = 0;  // number of immutable for compaction cells
   /**
-   * Types of Compaction
+   * Types of actions to be done on the pipeline upon MemStoreCompaction invocation
    */
-  private enum Type {
+  private enum ActionUponInvocation {
```

Comment:
ActionUponInvocation --> Action

In general, I'd suggest to avoid the Merge vs Compact terminology, this is confusing. Can we just distinguish between Index Compaction vs Data Compaction, like in the spec?